code-challenges/tree-intersection/tree-intersection.test.js

```diff
@@ -0,0 +1,62 @@
+
+class Node {
+  constructor(value = null, left = null, right = null)  {  
+    this.value = value;
+    this.left = left;
+    this.right = right;
+  }
+}
+
+class Tree {
+  constructor(root = null) {
+    this.root = root;
+  }
+
+}
+
+function ifATreeFallsInTheWoods(tree1, tree2) {
+  const allMap = new Map()
+  const collisionsMap = new Map()
+  preOrder(tree1.root, allMap, collisionsMap)
+  preOrder(tree2.root, allMap, collisionsMap)
+  return [...collisionsMap.keys()];
+  
+}
+
+function preOrder(current, allMap, collisionsMap) {
+  // console.log(current)
+  
+  if (!current) return
+  
+  //if the map already has the value, add the value to the collisions map. 
+  if (allMap.has(current.value)) {
+    collisionsMap.set(current.value, true)
```

Comment:
Tyler,
What if tree1 has duplicates? For example, tree1 is 1->1, tree2 is null. What's your output? You can make an assumption that trees do not have duplicates. But can you modify your code to handle this situation?